script.js

```diff
@@ -0,0 +1,385 @@
+    textarea.focus();
+    if (event.target.dataset.code === "CapsLock") return;
+    removeStyle();
+    identifyKey(event.target.dataset, "up");
+    addKeysOnKeyboard();
+  });
+
+  document.addEventListener("keydown", function (event) {
+    if (event.repeat) return;
+    event.preventDefault();
+    setStyle(document.querySelector(`.${KEY_CODE[event.code][0]}`), true);
+    identifyKey(event, "down");
+    checkLang();
+    addKeysOnKeyboard();
+  });
+
+  document.addEventListener("keyup", function (event) {
+    if (event.repeat) return;
+    if (event.key === "CapsLock") return;
+    setStyle(document.querySelector(`.${KEY_CODE[event.code][0]}`), false);
+    identifyKey(event, "up");
+    checkLang();
+    addKeysOnKeyboard();
+  });
+}
+
+function setStyle(code, bool) {
+  if (bool) code.classList.add("active");
+  else code.classList.remove("active");
+}
+
+function removeStyle() {
+  LastPress.classList.remove("active");
+}
+
+function identifyKey(event, updown) {
+  if (event.code === "ShiftLeft") {
+    if (shiftCaseLeft) shiftCaseLeft = false;
+    else shiftCaseLeft = true;
+  }
+
+  if (event.code === "AltLeft") {
+    if (altCaseLeft) altCaseLeft = false;
+    else altCaseLeft = true;
+  }
+
+  if (event.code === "CapsLock") {
+    if (isCapsLock === true) {
+      setStyle(document.querySelector(`.${KEY_CODE[event.code][0]}`), false);
+      document.querySelector(".CapsLock").classList.remove("caps-active");
+    } else document.querySelector(".CapsLock").classList.add("caps-active");
+    capsButtonIsPress();
+  }
+
+  if (
+    event.key === "shift" ||
+    event.key === "rShift" ||
+    event.key === "Shift"
+  ) {
+    capsButtonIsPress();
+    return;
+  }
+
+  if (updown === "down") {
+    if (event.code === "Tab") {
+      tabCase();
+    }
+
+    if (event.code === "Backspace") {
+      backspaceButton();
+    }
+
+    if (event.code === "Delete") {
+      deleteButton();
+    }
+
+    if (isShift === false && localStorage.lang === "ENG") {
+      if (KEY_CODE[event.code][1] !== undefined) {
+        printKeyCode(KEY_CODE[event.code][1]);
+      }
+    }
+
+    if (isShift === true && localStorage.lang === "ENG") {
+      if (KEY_CODE[event.code][2] !== undefined) {
+        printKeyCode(KEY_CODE[event.code][2]);
+      }
+    }
+
+    if (isShift === false && localStorage.lang === "RUS") {
+      if (KEY_CODE[event.code][3] !== undefined) {
+        printKeyCode(KEY_CODE[event.code][3]);
+      }
+    }
+
+    if (isShift === true && localStorage.lang === "RUS") {
+      if (KEY_CODE[event.code][4] !== undefined) {
+        printKeyCode(KEY_CODE[event.code][4]);
+      }
+    }
+  }
+}
+
+function printKeyCode(key) {
+  let arr = textarea.value.split("");
+  let pos = textarea.selectionStart;
+  arr.splice(pos, 0, key);
+  textarea.value = arr.join("");
+  textarea.selectionStart = textarea.selectionEnd = pos + 1;
+}
+
+function checkLang() {
+  if (changeLangUpKeyFlag) {
```

Comment:
change to `changeLangUpKeyFlag  && altCaseLeft === false && shiftCaseLeft === false`